target/linux/ath79/image/generic.mk

```diff
@@ -418,6 +439,21 @@ define Device/avm_fritzdvbc
 endef
 TARGET_DEVICES += avm_fritzdvbc
 
+define Device/avm_fritzx490-wasp
+  SOC := qca9558
+  DEVICE_VENDOR := AVM
+  DEVICE_MODEL := FRITZ!Box 3490/5490/7490 WASP (Wireless Assist)
+  KERNEL := kernel-bin | append-dtb | relocate-kernel | lzma-no-dict | \
+      avm-wasp-loader | pad-to 4096 | pad-extra 208 | avm-wasp-checksum x490
+  LOADER_TYPE := bin
+  KERNEL_INITRAMFS := $$(KERNEL)
+  DEVICE_PACKAGES := kmod-ath10k-ct ath10k-firmware-qca988x-ct-full-htt \
```

Comment:
Ok, I guess I should have thought about this earlier (the comment above that waffles about IMAGE/KERNEL_SIZE  makes no sense now.)

This DEVICE_PACKAGES will sadly not work with initramfs in the way you intend to. The OpenWrt builders use 
([CONFIG_TARGET_MULTI_PROFILE=y + CONFIG_TARGET_PER_DEVICE_ROOTFS=y](https://downloads.openwrt.org/snapshots/targets/ath79/generic/config.buildinfo)) to build all the images at once.

Hence, it compiles one shared initramfs for the whole sub-target and because of that it just includes what packages
are specified in DEFAULT_PACKAGES in the main ath79 Makefile `target/linux/ath79/Makefile` (kmod-gpio-button-hotplug swconfig kmod-ath9k uboot-envtools) and  subtarget `target/linux/ath79/generic/target.mk` (wpad-basic-wolfssl)). It will ignore the DEVICE_PACKAGES you set here. So you can't count on people downloading the builder's image to have your selection of (ath10k-ct/ath10k-firmware-qca988x, wpad). :disappointed: 

(A possible alternative to initramfs would be to use the older [initrd](https://www.kernel.org/doc/html/latest/admin-guide/initrd.html). This would make it possible to have something with your own packages in a filesystem that can be shipped together with the kernel)

Reply:
Too bad. Or I leave it as it is and add a comment to the wiki, to use an online image builder and add a list of packages manually?! And/Or should I remove the device packages here?

Reply:
initrd could be viable. You'll need to enable both CONFIG_BLK_DEV_INITRD + CONFIG_BLK_DEV_RAM in the ath79 target. Chances are, that you can just add them both to the generic config and nobody will care (i.e.: no device needs to be disabled because of kernel size issues). Then you have to attach the packed+cpio'd initrd and tell the kernel about its location. I think you can do this through the device-tree ([linux,initrd-start + linux,initrd-end](https://www.kernel.org/doc/Documentation/devicetree/bindings/chosen.txt)).